sensu/plugins/metrics-keystone.py

```diff
@@ -0,0 +1,63 @@
+#/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+
+import time
+from collections import namedtuple
+
+import shade
+
+OK = 0
+WARNING = 1
+CRITICAL = 2
+UNKNOWN = 3
+
+
+class CloudMetrics(object):
+    def __init__(self):
+        self.cloud = shade.openstack_cloud()
+
+    def num_projects(self):
+        return len(self.cloud.keystone_client.tenants.list())
+
+    def num_users(self):
+        return len(self.cloud.keystone_client.users.list())
+
+    def users_per_project(self):
+        UsersPerProject = namedtuple('UsersPerProject', 'proj_id, num_users')
+        return [
+            UsersPerProject(proj.id, len(proj.list_users()))
+            for proj in self.cloud.keystone_client.tenants.list()
```

Comment:
We've already done the expensive project listing, couldn't we cache that data somewhere to avoid a second API call here?  Additionally, when we get back all the users from num_users, do we get enough data back to show what project they are in? Could we avoid any further API calls here?